include/seqan3/io/alignment_file/detail.hpp

```diff
@@ -227,4 +231,151 @@ std::string get_cigar_string(alignment_type && alignment,
                             query_start_pos, query_end_pos, extended_cigar);
 }
 
+/*!\brief Transforms a cigar string into a vector of operation-count pairs (e.g. (M, 3)).
+ * \ingroup alignment_file
+ *
+ * \tparam soft_clipping_begin_type  The type of the soft_clipping_begin value; Must model std::Integral.
+ * \tparam soft_clipping_end_type    The type of the soft_clipping_end value; Must model std::Integral.
+ * \tparam cigar_view_type           The type of a single pass input view over the cigar string.
+ *
+ * \param[out] soft_clipping_begin  Set to the soft clipping at the beginning of the alignment.
+ * \param[out] soft_clipping_end    Set to the soft clipping at the end of the alignment.
+ * \param[in]  cigar_view           The single pass input view over the cigar string to parse.
+ *
+ * \returns A std::vector of operation-count pairs, e.g. (M, 3), that describe the alignment.
+ *
+ * \details
+ *
+ * For example, the view over the cigar string "1S4M1D2M2S" will set soft_clipping_begin to 1, soft_clipping_end to 2
+ * and return the vector "[(M,4), (D,1), (M,2)]".
+ */
+template <std::Integral soft_clipping_begin_type,
+          std::Integral soft_clipping_end_type,
+          std::ranges::InputRange cigar_view_type>
+std::vector<std::pair<char, size_t>> parse_cigar(soft_clipping_begin_type & soft_clipping_begin,
```

Comment:
Because the first parameter (soft clipped bases at the start) is also the OFFSET value read from file into the file record. The type of the OFFSET field can be specified via the file type traits so it is generic and without passing it as a parameter I cannot know the type in this function, and casting afterwards might lose information.